### ISSUE TITLE
Write tests for functions that call `send`

### ISSUE THREAD

**Issue by Fethbita** [Fethbita/emrtd#2]
Using test vectors and by mocking the [`send`](https://github.com/Fethbita/emrtd/blob/241c0d9eb2de5d98a8c94c9ac62d686bf02c998d/src/lib.rs#L2666-L2863) or `card.transmit` functions, unit tests for functions that call the `send` function can be written. However mocking a `pcsc::Card` proved to be difficult, an issue is created in `pcsc` GitHub: https://github.com/bluetech/pcsc-rust/issues/50.

The functions that can get unit tests after this are:
* [`read_data_from_ef`](https://github.com/Fethbita/emrtd/blob/241c0d9eb2de5d98a8c94c9ac62d686bf02c998d/src/lib.rs#L3093-L3203),
* [`establish_bac_session_keys`](https://github.com/Fethbita/emrtd/blob/241c0d9eb2de5d98a8c94c9ac62d686bf02c998d/src/lib.rs#L3205-L3346),
* Protected APDU creation or parsing of the `send` function,
* [`process_secure_rapdu`](https://github.com/Fethbita/emrtd/blob/241c0d9eb2de5d98a8c94c9ac62d686bf02c998d/src/lib.rs#L2865-L3003).

**Comment by dishmaker** [comment]
I think you can create a trait like:
```rust
pub trait MrtdCard {

    /// Returns answer to reset
    fn atr(&self) -> Result<Vec<u8>>

    /// Returns R-APDU including 2 bytes SW
    fn transmit(&self, c_apdu: &[u8]) -> Result<Vec<u8>>


}
```

so that your wrapper takes a generic param:

```rust

pub struct EmrtdComms<C: MrtdCard> {
    // ...
}

impl<C: MrtdCard> EmrtdComms<C> {
    // ...

}
```

**Comment by Fethbita** [comment]
Thank you for the tip @dishmaker. Implemented this solution in #4, also implementing a trait based solution for the Rng.
